python/tk_hiero_export/sg_shot_processor.py

```diff
@@ -256,6 +194,73 @@ def _get_tags(self, items):
         tags = [tag for tag in tags if "Nuke Project File" not in tag.name()]
         return tags
 
+class ShotgunShotProcessor(ShotgunHieroObjectBase, FnShotProcessor.ShotProcessor):
+    """
+    Adds hook functionality to the built in Shot processor.
+    """
+    def __init__(self, preset, submission=None, synchronous=False):
+        FnShotProcessor.ShotProcessor.__init__(self, preset, submission, synchronous)
+
+        # Call pre processor hook here to make sure it happens pior to any 'hook_resolve_custom_strings'.
+        # The order if execution is basically [init processor, resolve user entries, startProcessing].
+        self.app.execute_hook("hook_pre_export", processor=self)
```

Comment:
I think we should be pragmatic about it - it's a big change - at the end of the day Hiero broke its own compatibility, that's why we have to do these refactors :) - so I think it's fair enough that there is a risk that we'll do the same to hooks given that in a sense we are just forwarding the objects on to the user. 

I think we should highlight it in the release notes and give people a heads up about it in support. And then we'll try to help people refactor their hooks if needed!